test/e2e/common/secrets.go

```diff
@@ -196,15 +222,16 @@ func doSecretE2EWithoutMapping(f *framework.Framework, defaultMode *int32) {
 
 	pod := &api.Pod{
 		ObjectMeta: api.ObjectMeta{
-			Name: "pod-secrets-" + string(uuid.NewUUID()),
+			Name:      "pod-secrets-" + string(uuid.NewUUID()),
+			Namespace: f.Namespace.Name,
 		},
 		Spec: api.PodSpec{
 			Volumes: []api.Volume{
 				{
 					Name: volumeName,
 					VolumeSource: api.VolumeSource{
 						Secret: &api.SecretVolumeSource{
-							SecretName: name,
+							SecretName: secret.Name,
```

Comment:
It seems confusing to have `name` and `secret.Name`, why don't we have just one? (i.e, name = secretName if not empty and the same random value otherwise)